pulsar-transformations/src/test/java/com/datastax/oss/pulsar/functions/transforms/jstl/JstEvaluatorTest.java

```diff
@@ -46,6 +48,47 @@ void testFunctions(String expression, TransformContext context, Object expectedV
         new JstlEvaluator<>(String.format("${%s}", expression), Object.class).evaluate(context));
   }
 
+  @Test
+  void testNowFunction() {
+    Record<GenericObject> primitiveStringRecord =
+        new Utils.TestRecord<>(
+            Schema.STRING,
+            AutoConsumeSchema.wrapPrimitiveObject("test-message", SchemaType.STRING, new byte[] {}),
+            "header-key");
+    TransformContext primitiveStringContext =
+        new TransformContext(
+            new Utils.TestContext(primitiveStringRecord, new HashMap<>()),
+            primitiveStringRecord.getValue().getNativeObject());
+
+    long expectedNow = Instant.now().toEpochMilli();
+    long actualNow =
+        new JstlEvaluator<Long>("${fn:now()}", long.class).evaluate(primitiveStringContext);
+
+    assertTrue(actualNow >= expectedNow);
+    assertTrue(actualNow < expectedNow + 100L);
+  }
+
+  @Test
+  void testDateAddFunctionsNow() {
+    Record<GenericObject> primitiveStringRecord =
+        new Utils.TestRecord<>(
+            Schema.STRING,
+            AutoConsumeSchema.wrapPrimitiveObject("test-message", SchemaType.STRING, new byte[] {}),
+            "header-key");
+    TransformContext primitiveStringContext =
+        new TransformContext(
+            new Utils.TestContext(primitiveStringRecord, new HashMap<>()),
+            primitiveStringRecord.getValue().getNativeObject());
+
+    long expected = Instant.now().plusSeconds(-3333).toEpochMilli();
+    long actual =
+        new JstlEvaluator<Long>("${fn:dateadd(fn:now(), -3333, 'seconds')}", long.class)
```

Comment:
I think it would be nice to be able to do pass an ISO8601 duration: https://en.m.wikipedia.org/wiki/ISO_8601#Durations. 
Jackson has good support for deserializing Durations.

Reply:
Jackson may not even be needed. It's probably as simple as Duration.parse() : https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-

Reply:
+1, will add in a separate PR: https://github.com/datastax/pulsar-transformations/issues/28